app/controllers/concerns/commented.rb

```diff
@@ -2,14 +2,36 @@ module Commented
   extend ActiveSupport::Concern
 
   included do
-    helper_method :commentable, :comments
+    helper_method :commentable, :comments, :mentions_builder
   end
 
   def commentable
     instance_variable_get("@#{controller_name.singularize}")
   end
 
   def comments
-    @comments ||= commentable.comments.includes(:user)
+    @comments ||= commentable&.comments&.includes(:user)
+  end
+
+  def mentioned_users(comments)
+    # Match any non whitespace that starts with an @ has another @
+    emails = comments.inject([]) do |collection, comment|
+      (collection + comment.scan(/@(\S*@\S*)/)).flatten.uniq
+    end
+
+    current_project.authors.where(email: emails)
+  end
+
+  def mentions_builder(comments)
```

Comment:
I'd really like to write some test code for this with some email edge cases etc but the codebase doesn't do controller specs, let alone concerns for controllers. If anyone has a good idea where a place for testing this could go let me know and I'd cook something up.